### ISSUE TITLE
Hierarchies list question/bug - edition mode - left filter

### ISSUE THREAD

**Issue by FabianCretton** [visualize-admin/visualization-tool#822]
**Describe the bug**
I am working on a new hierarchy and see odd things in the display.

As this hierarchy is generated by a script, I am trying here to understand if:
- my generated hierarchy has problems (RDF)
- I do not fully understand how Visualize is using the hierarchy
- There is a display bug in Visualize

Cube: "fab Foreign Trade test 2" 
Environment TEST
In visualize, [here](https://test.visualize.admin.ch/en/browse/dataset/https%3A%2F%2Fenvironment.ld.admin.ch%2Ffoen%2Ffab_foreign_trade_test_2%2F3?previous=%7B%22includeDrafts%22%3Atrue%2C%22order%22%3A%22SCORE%22%2C%22search%22%3A%22fab%22%7D&dataSource=Test)

Here is an example: the node "WK '1' ROUND WOOD (WOOD IN THE ROUGH)" has a child "List of Zolltarifs", which has 43 children.
But the filter on the left "Tariff or Basket ID" displays only 4 children:
![image](https://user-images.githubusercontent.com/6167669/199920180-8887eea9-90dc-4d4b-87db-513f09c479a9.png)

The filter on the right, when choosing a color, does display all children:
![image](https://user-images.githubusercontent.com/6167669/199920232-e32007fc-cc4e-4a6a-b59b-ab3616ff2679.png)

Those 43 children are visible when [displaying the hierarchy in the Cube Creator](https://test.cube-creator.lindas.admin.ch/app/hierarchies/dimension!!hierarchy!!hierarchy-swiss-foreign-trade-baskets+zolltarif-(on-bafu-helpers)). I suppose that the RDF is thus correct.

I see "missing" nodes in different parts of the hierarchy, I could give more examples if needed.

**Comment by FabianCretton** [comment]
@ptbrowne salut Patrick, aurais-tu quelques infos à ce sujet ? Est-ce une incompréhension de ma part, une erreur dans les données, un bug de Visualize ? Merci d'avance

**Comment by ptbrowne** [comment]
Hi @FabianCretton , I'll have a look tomorrow, it seems we are effectively not returning all the nodes but I do not know exactly why and right now I get 503 from test.lindas.admin.ch.

<details>
  <summary>Technical details</summary>

  ```
query DimensionHierarchy($sourceType: String!, $sourceUrl: String!, $locale: String!, $cubeIri: String!, $dimensionIri: String!) {
  dataCubeByIri(
    iri: $cubeIri
    sourceType: $sourceType
    sourceUrl: $sourceUrl
    locale: $locale
  ) {
    dimensionByIri(
      iri: $dimensionIri
      sourceType: $sourceType
      sourceUrl: $sourceUrl
    ) {
      ...hierarchyMetadata
      __typename
    }
    __typename
  }
}

fragment hierarchyMetadata on Dimension {
  hierarchy(sourceType: $sourceType, sourceUrl: $sourceUrl) {
    ...hierarchyValueFields
    children {
      ...hierarchyValueFields
      children {
        ...hierarchyValueFields
        children {
          ...hierarchyValueFields
          __typename
        }
        __typename
      }
      __typename
    }
    __typename
  }
}

fragment hierarchyValueFields on HierarchyValue {
  value
  dimensionIri
  depth
  label
  hasValue
}

```

Variables

```
{
  "cubeIri": "https://environment.ld.admin.ch/foen/fab_foreign_trade_test_2/4",
  "dimensionIri": "https://environment.ld.admin.ch/foen/fab_foreign_trade_test_2/tnorwarenkoerbeid",
  "sourceType": "sparql",
  "sourceUrl": "https://test.lindas.admin.ch/query",
  "locale": "en"
}
```
</details>

**Comment by ptbrowne** [comment]
This hierarchy seems a bit strange in the sense that there are children that are reached through multiple chains of parents. It would mean that in the left panel select, an option could appear multiple times. Is this wanted @FabianCretton , can you explain a bit more what is this data ?

Attached is a rendition of the hierarchy, where we see that some children on the right have multiple incoming arrows = multiple parents. This is a usecase that was not foreseen, it could be changed on visualize quite easily, but I just want to make sure that this is normal.



![Image](https://user-images.githubusercontent.com/465582/200842201-168ac2fc-e42c-40e3-a954-7bc11dbfbac3.svg)

**Comment by FabianCretton** [comment]
Thank you for the information Patrick.
And yes, it is currently on purpose that a resource could have different parents.
Those leaf "resources" are tarifs.
The first case is that we have different baskets, and one tarif can be found in more than one basket.
The other case is that the baskets are in a hierarchy, and if a tarif is in a basket of 'level 2', then we also want to show that it is in the parent basket of level 1.
This was my way to do it, but we can discuss about it of course, if it doesn't make sense.

**Comment by ptbrowne** [comment]
It seems to me that the 1st case is OK, while the second case is more problematic as the total of the children would be over 100% right ?

Example:

```
A 
  B
  C
     D
  D
```

B + C + D > 100% and I think it could be problematic. I think here, it should be handled through another mean that the parent relationship, since this is wanted only for "viewing".

@l00mi What's your thoughts on this ?

**Comment by FabianCretton** [comment]
Yes, I think you are right, currently it is more for viewing what is inside a basket at a specific level.
But the total value, calculated to create the input CSV, is correct at each level (I think that Visualize still do not calculate agregate and they must be provided in the dataset, right @ptbrowne ?)
A special case here is that we talk about tarifs (the real values), which are grouped in baskets.
The baskets do have a hierarchy. 
But then, a basket can contain other baskets, but also directly a tarif that is not in a sub-basket.
For instance: 
- basket 1 is the parent of basket 1.1 and basket 1.2
- basket 1.1 contains tarif A
- basket 1.2 contains tarif B
- basket 1 contains tarif A (from 1.1), tarif B (from 1.2) and tarif C (directly here)
Is that ok ?
We might want to involve Achim Schafer in this discussion.
